src/rpc/cli.cpp

```diff
@@ -134,53 +120,73 @@ char * dupstr (const char* s) {
    return (r);
 }
 
-char* my_generator(const char* text, int state)
+/****
+ * @brief loop through list of commands, attempting to find a match
+ * @param token what the user typed
+ * @param match
+ * @returns the full name of the command or NULL if no matches found
+ */
+static char *my_rl_complete(char *token, int *match)
 {
-   static int list_index, len;
-   const char *name;
-
-   if (!state) {
-      list_index = 0;
-      len = strlen (text);
-   }
-
-   auto& cmd = cli_commands();
-
-   while( list_index < cmd.size() ) 
-   {
-      name = cmd[list_index].c_str();
-      list_index++;
-
-      if (strncmp (name, text, len) == 0)
-         return (dupstr(name));
```

Comment:
With this, the dupstr() function will no longer be used an should be removed as well.

Reply:
Removed